tests/apollo/test_skvbc_checkpoints.py

```diff
@@ -123,6 +123,106 @@ async def test_checkpoint_propagation_after_crashed_replicas_comeback(self, bft_
 
             self.assertEqual(checkpoint_stale, checkpoint_after_primary)
 
+    @with_trio
+    @with_bft_network(start_replica_cmd)
+    async def test_checkpoint_propagation_after_crashed_replicas_comeback_after_vc(self, bft_network):
+        """
+        Here we bring down a total of f - 1 replicas, create checkpoint, trigger
+        view change by bringing down the primary (total f replicas down).
+        We then verify checkpoint propagation in the new view.
+
+        1) Given a BFT network, we make sure all nodes are up
+        2) Crash f - 1 replicas, excluding the initial primary, and next view primary
+        3) Send sufficient number of client requests to trigger checkpoint protocol
+        4) Make sure checkpoint is created
+        5) Crash the initial primary & send a batch of write requests
+        6) Verify view change has occurred
+        7) Verify checkpoint propagation to current primary in the new view
+        8) Bring the f replicas up
+        9) Verify checkpoint propagation to a stale node of the f replicas brought up
+        """
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+
+        n = bft_network.config.n
+        f = bft_network.config.f
+        c = bft_network.config.c
+
+        self.assertEqual(len(bft_network.procs), n,
+                         "Make sure all replicas are up initially.")
+
+        initial_primary = await bft_network.get_current_primary()
+
+        expected_next_primary = initial_primary + 1
+
+        checkpoint_init_primary_before = await bft_network.wait_for_checkpoint(replica_id=initial_primary)
+
+        # crashing f-1 replicas, initial_primary is crashed after checkpoint creation to trigger vc
+        crashed_replicas = await self._crash_replicas(
+            bft_network=bft_network,
+            nb_crashing=f - 1,
+            exclude_replicas={expected_next_primary, initial_primary}
+        )
+        self.assertFalse(expected_next_primary and initial_primary in crashed_replicas)
```

Comment:
The correct way to do this would be:
`self.assertTrue({initial_primary, expected_next_primary}.issubset(crashed_replicas))`

Reply:
Thanks, updated.